tests/timer/test_Timer.cpp

```diff
@@ -0,0 +1,212 @@
+#include "gtest/gtest.h"
+
+#include "Kokkos_Core.hpp"
+
+#include "kokkos-utils/timer/Timer.hpp"
+
+/**
+ * @file
+ *
+ * @addtogroup unittests
+ *
+ * Timer
+ * -----
+ *
+ * This group of tests check the behavior of @ref Kokkos::utils::timer::Timer.
+ */
+
+using execution_space = Kokkos::DefaultExecutionSpace;
+
+namespace Kokkos::utils::tests::timer
+{
+
+using namespace Kokkos::utils::timer;
+
+using timer_t = Timer<execution_space>;
+
+class TimerTest : public ::testing::Test
+{
+public:
+    void SetUp() override {
+        this->exec = Kokkos::Experimental::partition_space(execution_space{}, 1)[0];
+    }
+
+protected:
+    execution_space exec {};
+};
+
+/**
+ * @test Check that @ref Kokkos::utils::timer::Timer has a type alias @ref Kokkos::utils::timer::Timer::event_t that satisfies
+ *       the concept @ref Kokkos::utils::timer::IsEvent.
+ */
+TEST(Timer, event_t_type_alias_is_event)
+{
+    static_assert(IsEvent<typename timer_t::event_t, execution_space, milliseconds>);
+}
+
+/**
+ * @test Check that @ref Kokkos::utils::timer::Timer reports it's invalid when neither @ref Kokkos::utils::timer::Timer::reset
+ *       nor @ref Kokkos::utils::timer::Timer::stop has been called.
+ */
+TEST_F(TimerTest, invalid_if_not_reset_and_not_stopped)
+{
+    const timer_t timer;
+    ASSERT_FALSE(timer.is_valid());
+}
+
+/**
+ * @test Check that @ref Kokkos::utils::timer::Timer reports it's invalid when @ref Kokkos::utils::timer::Timer::reset has been called,
+ *       but @ref Kokkos::utils::timer::Timer::stop has not.
+ */
+TEST_F(TimerTest, invalid_if_not_stopped)
+{
+    timer_t timer;
+    timer.reset(this->exec);
+    ASSERT_FALSE(timer.is_valid());
+}
+
+/**
+ * @test Check that @ref Kokkos::utils::timer::Timer reports it's invalid when @ref Kokkos::utils::timer::Timer::reset has not been called,
+ *       but @ref Kokkos::utils::timer::Timer::stop has.
+ */
+TEST_F(TimerTest, invalid_if_not_reset)
+{
+    timer_t timer;
+    timer.stop(this->exec);
+    ASSERT_FALSE(timer.is_valid());
+}
+
+/**
+ * @test Check that @ref Kokkos::utils::timer::Timer reports it's valid when @ref Kokkos::utils::timer::Timer::reset and
+ *       @ref Kokkos::utils::timer::Timer::stop have been called.
+ */
+TEST_F(TimerTest, valid_if_reset_and_stopped)
+{
+    timer_t timer;
+    timer.reset(this->exec);
+    timer.stop (this->exec);
+    ASSERT_TRUE(timer.is_valid());
+}
+
+/**
+ * @test Check @ref Kokkos::utils::timer::Timer by verifying that the elapsed time is less
+ *       than the time measured by a @c Kokkos::Timer wrapped outside of it.
+ */
+TEST_F(TimerTest, start_stop_elapsed)
+{
+    using timer_external_t = Kokkos::Timer;
+
+    using view_t = Kokkos::View<double*, execution_space>;
+
+    timer_external_t timer_external;
```

Comment:
No need for such a *using* if used once, rightaway.
```suggestion
    using view_t = Kokkos::View<double*, execution_space>;

    Kokkos::Timer timer_external;
```

Reply:
OK. Longer version not needed indeed!